### PR TITLE
Use Webpack v5.x for asset bundling

### LINKED ISSUE #4020: Upgrading webpack from 4 to 5
### Environment

phoenix 1.5.5
elixir      1.10.3-otp-23 
erlang   23.0.1 
nodejs  12.18.4  
npm      6.14.6
OSX 10.15.6 (19G2021)
Webpack 5
```
% mix phx.server
Compiling 59 files (.ex)
Generated morphic_pro app
[info] [alarm_handler: {:set, {:system_memory_high_watermark, []}}]
[info] Running MorphicProWeb.Endpoint with cowboy 2.8.0 at 0.0.0.0:4001 (https)
[info] Access MorphicProWeb.Endpoint at https://localhost:4001
[webpack-cli]
TypeError: Cannot read property 'tap' of undefined
[error] Task MorphicProWeb.Endpoint started from #PID<0.1043.0> terminating
** (stop) :watcher_command_error
    (phoenix 1.5.5) lib/phoenix/endpoint/watcher.ex:37: Phoenix.Endpoint.Watcher.watch/3
    (elixir 1.10.3) lib/task/supervised.ex:90: Task.Supervised.invoke_mfa/2
    (stdlib 3.13) proc_lib.erl:226: :proc_lib.init_p_do_apply/3
Function: &Phoenix.Endpoint.Watcher.watch/3
    Args: ["node", ["node_modules/webpack/bin/webpack.js", "--config", "webpack.dev.config.js", "--color", "--bail"], [cd: "/Users/joshchernoff/Dev/morphic.pro/assets"]]
```
Looks like watch is broken with the webpack 5 upgrade. 

Also, note `--watch-stdin` is no longer an option

### PR DESCRIPTION
**Pull request by acconrad**
Resolves #4020 by extending and fixing outstanding issues from #4054 

## Summary of changes

* Ensures asset bundling works at least for LTS versions of Node for new projects
* Upgrades dependencies to the latest secure versions of the dependencies required to build and deploy Phoenix projects
* Removes now unnecessary plugins for Webpack (e.g. terser plugin)
* Updates Webpack config to match naming and conventions for v5, including support for JS, CSS, images, fonts, and other assets
* Updates Phoenix configs to reflect correct Webpack conventions and prevent possible deploy warnings from missing properties

### CODE REVIEW

**Comment by masterkain** [comment]
I applied to pr, it is complaning about `glob` for some reason

```
iex(1)> [webpack-cli] Failed to load '/Users/kain/Sites/bscharts/assets/webpack.config.js' config
[webpack-cli] Error: Cannot find module 'glob'
Require stack:
- /Users/kain/Sites/bscharts/assets/webpack.config.js
- /Users/kain/Sites/bscharts/assets/node_modules/webpack-cli/lib/webpack-cli.js
- /Users/kain/Sites/bscharts/assets/node_modules/webpack-cli/lib/bootstrap.js
- /Users/kain/Sites/bscharts/assets/node_modules/webpack-cli/bin/cli.js
- /Users/kain/Sites/bscharts/assets/node_modules/webpack/bin/webpack.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:880:15)
    at Function.Module._load (internal/modules/cjs/loader.js:725:27)
    at Module.require (internal/modules/cjs/loader.js:952:19)
    at require (/Users/kain/Sites/bscharts/assets/node_modules/v8-compile-cache/v8-compile-cache.js:159:20)
    at Object.<anonymous> (/Users/kain/Sites/bscharts/assets/webpack.config.js:2:14)
    at Module._compile (/Users/kain/Sites/bscharts/assets/node_modules/v8-compile-cache/v8-compile-cache.js:192:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14)
    at Module.require (internal/modules/cjs/loader.js:952:19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/Users/kain/Sites/bscharts/assets/webpack.config.js',
    '/Users/kain/Sites/bscharts/assets/node_modules/webpack-cli/lib/webpack-cli.js',
    '/Users/kain/Sites/bscharts/assets/node_modules/webpack-cli/lib/bootstrap.js',
    '/Users/kain/Sites/bscharts/assets/node_modules/webpack-cli/bin/cli.js',
    '/Users/kain/Sites/bscharts/assets/node_modules/webpack/bin/webpack.js'
  ]
}
```

**Comment by masterkain** [comment]
to make this work I had to:

- `cd assets`
- `npm i glob --save-dev`
- `npm install css-minimizer-webpack-plugin --save-dev`

**Comment by aj-foster** [comment]
Re: `glob`, see also [this comment](https://github.com/phoenixframework/phoenix/pull/4054#issuecomment-821402239).

**Comment by acconrad** [comment]
@masterkain @aj-foster fixed

File: installer/templates/phx_assets/package.json [hunk 1]
```diff
@@ -1,26 +1,30 @@
 {
   "private": true,
+  "engines": {
+    "node": ">= 14.16.1"
+  },
   "scripts": {
-    "deploy": "webpack --mode production",
-    "watch": "webpack --mode development --watch"
+    "deploy": "webpack --mode=production",
+    "watch": "webpack --mode=development --watch"
```
Comment (totaltrash) [inline installer/templates/phx_assets/package.json:8]: this also needs `--watch-options-stdin`?

Comment (cw789) [reply installer/templates/phx_assets/package.json:8]: No. It's fine here without.
`mix phx.server` does not use this command directly (`npm run watch`).

**Comment by chrismccord** [comment]
Thank you for all the leg work on this! @bcardarella has taken care of webpack v5 support in #4337 ❤️❤️❤️🐥🔥

**Pull request closed without merging**